Sources/NIOSSL/PosixPort.swift

```diff
@@ -100,8 +100,8 @@ internal func wrapErrorIsNullReturnCall<T>(where function: String = #function, _
 // MARK:- Our functions
 internal enum Posix {
     @inline(never)
-    internal static func fopen(file: UnsafePointer<CChar>, mode: UnsafePointer<CChar>) throws -> FILEPointer {
-        return try wrapErrorIsNullReturnCall {
+    internal static func fopen(file: String, mode: String) throws -> FILEPointer {
```

Comment:
Can we add an indirection here? The `String` to `char *` pointer transformation potentially dirties up `errno`, so we don't want it to implicitly happen within the body of `wrapErrorIsNullReturnCall`.

Reply:
Done in https://github.com/apple/swift-nio-ssl/pull/375/commits/199bdac198e98320dacf1ad8b9c8d2d0f0c0b53f